LandSandBoat/server
# [SMN] Correct MP cost for BP abilities, add Shining Ruby to pet_skills.sql

**_I affirm:_**

## What does this pull request do?


This PR corrects a issue with Blood Pact ability MP consumption. Currently, when SMN uses a Blood Pact type ability it will consume MP per targeted entity. Adding the "action" argument will ensure the MP is only consumed if the target of the ability is the initial target rather than each target.

Additional changes:
- Better align arguments to match onPetAbility order
- Use "summoner" argument to determine the pets master
- Correct missing Shining Ruby pet ability for Carbuncle

## Steps to test these changes


Tested changed SMN abilities to ensure it consumes the correct MP for the initial target of BP Ward and Rage abilities.
Tested the job ability "Apogee" to ensure x1.5 MP consumption
